src/rust/rust-proto-new/src/serde_impl.rs

```diff
@@ -0,0 +1,36 @@
+use bytes::{
+    Bytes,
+    BytesMut,
+};
+use prost::Message;
+
+use crate::{
+    type_url,
+    SerDe,
+    SerDeError,
+};
+
+pub trait HasAssociatedProto<T> {
+    type Proto: From<T> + Message + Default + TryInto<T, Error = SerDeError>;
```

Comment:
Definitely want to make sure this doesn't export `prost::Message` from this crate. If `pub(crate)` ensures this then 👍

Reply:
done (afaik)